effort_controllers/src/joint_group_position_controller.cpp

```diff
@@ -120,14 +120,26 @@ namespace effort_controllers
     return true;
   }
 
+  void JointGroupPositionController::starting(const ros::Time& time)
+  {
+    std::vector<double> current_positions(n_joints_, 0.0);
+    for (std::size_t i = 0; i < n_joints_; ++i)
+    {
+      current_positions[i] = joints_[i].getPosition();
+      enforceJointLimits(current_positions[i], i);
+      pid_controllers_[i].reset();
+    }
+    commands_buffer_.writeFromNonRT(current_positions);
```

Comment:
`starting()` is in the RT domain, so rather than calling `writeFromNonRT()`, you should write to the command buffer directly. (The realtime buffer lacks a proper writeFromRT method since it's really intended for writing from a single non-RT thread, reading from a single RT thead. I haven't worked through the accesses and how safe this aooriach is, but from these examples by people much smarter than me, it seems this is a safe solution in `starting()`)

Examples:
https://github.com/ros-controls/ros_controllers/blob/1fd8d25707e289700743925c93644b4295b1c486/position_controllers/src/joint_group_position_controller.cpp#L42-L50
https://github.com/ros-controls/ros_controllers/blob/1fd8d25707e289700743925c93644b4295b1c486/velocity_controllers/src/joint_group_velocity_controller.cpp#L42-L46

Reply:
Actually, looking at this again, I think the right solution is to just use `initRT()`. In practical senses, it'll be pretty much the same thing, but with less overhead and also less hackey.

I think those examples I linked should probably be updated to `initRT()` as well. I'll look into that.